tasks/shell_pcs/pcs-auth-pcs-0.10.yml

```
@@ -12,15 +12,25 @@
           addr={{ hostvars[node].ha_cluster.pcs_address | quote }}
         {% endif %}
       {% endfor %}
-      {% if __ha_cluster_qdevice_model == "net" and __ha_cluster_qdevice_host %}
-        {# prevent pcs error: Host name defined multiple times #}
-        {% if  __ha_cluster_qdevice_host not in __ha_cluster_all_node_names %}
-          {{ __ha_cluster_qdevice_host | quote }}
-          {% if __ha_cluster_qdevice_pcs_address %}
-            addr={{ __ha_cluster_qdevice_pcs_address | quote }}
-          {% endif %}
-        {% endif %}
-      {% endif %}
     stdin: "{{ ha_cluster_hacluster_password }}"
   run_once: true  # noqa: run_once[task]
   changed_when: true
+
+- name: Pcs auth for qdevice using pcs-0.10
+  command:
+    # Always auth all nodes to prevent possible corner cases with synchronizing
+    # pcs auth tokens in the cluster when not all nodes are auth-ed.
+    cmd: >
+      pcs host auth -u hacluster --
+      {{ __ha_cluster_qdevice_host | quote }}
+      {% if __ha_cluster_qdevice_pcs_address %}
+        addr={{ __ha_cluster_qdevice_pcs_address | quote }}
+      {% endif %}
+    stdin: "{{ ha_cluster_hacluster_qdevice_password |
+      d(ha_cluster_hacluster_password) }}"
+  when:
+    - __ha_cluster_qdevice_model == "net"
+    - __ha_cluster_qdevice_host
```

```suggestion
    - __ha_cluster_qdevice_host | length > 0
```
this is causing the failures on 2.9 - it doesn't know how to evaluate this in a "bare" context - but 2.15 does.
What is this testing for?  I think this wants to see if `__ha_cluster_qdevice_host` is a non-zero length string since it is set here - https://github.com/linux-system-roles/ha_cluster/blob/main/tasks/shell_pcs/check-and-prepare-role-variables.yml#L58 - it looks like if it wasn't set to a valid host, the value will be `''` the empty string.